### PR TITLE
Access to build resources for people outside of build group - strawman for discussion

### PR DESCRIPTION
**Pull request by mhdawson**
As discussed in last build working group meeting.  Initial
strawman for discussion

### CODE REVIEW

**Comment by mhdawson** [comment]
@nodejs/build any comments ?

**Comment by rvagg** [comment]
Not yet, will try and take a look soon @mhdawson, just a bit snowed under atm. Also, fwiw, your PR subject, description and the commit details all say that it's a "strawman for discussion" but not about what, I had to go to the actual changes to figure out what this was even about. Basic details that appear in GH notifications would be nice to help with prioritisation.

File: doc/process/special_access_to_build_resources.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+* Is the invidual a collaborator.  If so then access should be allowed
+  provided the first point is satisfied.
+* Length and consistency of involvement with the working group or the
+  Node.js community.
+* Consequences to the invidudal in case of mis-bahaviour.  For example,
+  would they potentially lose their job if they were reported as 
+  mis-behaving to their employer ?  Would being banned from involvement
+  in the Node.js community negatively affect them personally in some other
+  way ?
+* Are there collaborators who work with the individual and can vouge for
```
Comment (orangemocha) [inline doc/process/special_access_to_build_resources.md]: `vouge`. Did you mean `vouch` ?

File: doc/process/special_access_to_build_resources.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+temporary account on the specific machines.  
+
+If the considerations listed for ongoing access are satisfied, access can
+be granted after discussion in a issue on the repo after one additional
+lgtm from a build team member (1 build team member raises issue, second
+approves)
+
+In cases that warrant it, access can be granted to a machine where
+the individual does not meet the requirements for ongoing access, however
+in this case the machine should first be disconnected from the
```
Comment (orangemocha) [inline doc/process/special_access_to_build_resources.md]: We might want to call out that we should also ensure that any secrets (eg credentials) are deleted from the machine before giving access.

File: doc/process/special_access_to_build_resources.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+approves)
+
+In cases that warrant it, access can be granted to a machine where
+the individual does not meet the requirements for ongoing access, however
+in this case the machine should first be disconnected from the 
+build farm, the length of access should be limited to the minimum
+feasible and and the machine should be re-imaged once access has been
+revoked. 
+
+Access should be recorded in a way that we can preriodically review
```
Comment (orangemocha) [inline doc/process/special_access_to_build_resources.md]: `preriodically` => `periodically`

Comment (orangemocha) [inline doc/process/special_access_to_build_resources.md]: Suggestion: we could easily keep track of access in the readme of the secrets repo.

**Comment by orangemocha** [comment]
Left a few minor comments. It also needs a reworded commit message.
LGTM once those things are addressed.

File: doc/process/special_access_to_build_resources.md (hunk 1 above)
Comment (mhdawson) [inline doc/process/special_access_to_build_resources.md]: yes

File: doc/process/special_access_to_build_resources.md (hunk 2 above)
Comment (mhdawson) [inline doc/process/special_access_to_build_resources.md]: added

**Comment by mhdawson** [comment]
Updated to address @orangemocha's commends

**Comment by orangemocha** [comment]
LGTM

**Comment by joaocgreis** [comment]
LGTM

File: doc/process/special_access_to_build_resources.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
+# Introduction
+
+There are a number of cases were we would like to provide access to
+community machines and or jenkins jobs to people who are not part
+of the build working group.  Examples include:
```
Comment (jbergstroem) [inline doc/process/special_access_to_build_resources.md]: Nit: two spaces before "Examples"

File: doc/process/special_access_to_build_resources.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+* test working group team members so that they can add/debug tests to/on 
+  community test infrastructure.
+* benchmark working group members so they can add benchmark jobs and
+  or experiement with benchmarking infrastructure in advance of the
+  final configuration being added to ansible.
+* PR's authors so they can debug failures across platforms in cases
```
Comment (jbergstroem) [inline doc/process/special_access_to_build_resources.md]: Perhaps mention that this for Node.js collaborators only (?).

File: doc/process/special_access_to_build_resources.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+In these cases the following will be used to decide if ongoing access
+can be provided:
+
+* Does the scope and size of the need justify providing access.
+* Is the invidual a collaborator.  If so then access should be allowed
+  provided the first point is satisfied.
+* Length and consistency of involvement with the working group or the
+  Node.js community.
+* Consequences to the invidudal in case of mis-bahaviour.  For example,
```
Comment (jbergstroem) [inline doc/process/special_access_to_build_resources.md]: Nit: two spaces

File: doc/process/special_access_to_build_resources.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
+# Introduction
+
+There are a number of cases were we would like to provide access to
+community machines and or jenkins jobs to people who are not part
+of the build working group.  Examples include:
+
+* test working group team members so that they can add/debug tests to/on
```
Comment (jbergstroem) [inline doc/process/special_access_to_build_resources.md]: nit: test -> Test

File: doc/process/special_access_to_build_resources.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
+# Introduction
+
+There are a number of cases were we would like to provide access to
+community machines and or jenkins jobs to people who are not part
+of the build working group.  Examples include:
+
+* test working group team members so that they can add/debug tests to/on 
+  community test infrastructure.
+* benchmark working group members so they can add benchmark jobs and
```
Comment (jbergstroem) [inline doc/process/special_access_to_build_resources.md]: nit: benchmark -> Benchmark

File: doc/process/special_access_to_build_resources.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+individual when this supports the efforts of a working group/team. For
+example some members of the test team will be regularly adding tests
+and therefore the ability to create/configure tests jobs needs to be
+granted in an ongoing manner.  
+
+In these cases the following will be used to decide if ongoing access
+can be provided:
+
+* Does the scope and size of the need justify providing access.
+* Is the invidual a collaborator.  If so then access should be allowed
```
Comment (jbergstroem) [inline doc/process/special_access_to_build_resources.md]: Nit: two spaces.

File: doc/process/special_access_to_build_resources.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+  provided the first point is satisfied.
+* Length and consistency of involvement with the working group or the
+  Node.js community.
+* Consequences to the invidudal in case of mis-bahaviour.  For example,
+  would they potentially lose their job if they were reported as 
+  mis-behaving to their employer ?  Would being banned from involvement
```
Comment (jbergstroem) [inline doc/process/special_access_to_build_resources.md]: nit: spaces

File: doc/process/special_access_to_build_resources.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+granted in an ongoing manner.  
+
+In these cases the following will be used to decide if ongoing access
+can be provided:
+
+* Does the scope and size of the need justify providing access.
+* Is the invidual a collaborator.  If so then access should be allowed
+  provided the first point is satisfied.
+* Length and consistency of involvement with the working group or the
+  Node.js community.
```
Comment (jbergstroem) [inline doc/process/special_access_to_build_resources.md]: perhaps: Node.js working groups and/or community

File: doc/process/special_access_to_build_resources.md [hunk 12]
```diff
@@ -0,0 +1,77 @@
...
+* Consequences to the invidudal in case of mis-bahaviour.  For example,
+  would they potentially lose their job if they were reported as 
+  mis-behaving to their employer ?  Would being banned from involvement
+  in the Node.js community negatively affect them personally in some other
+  way ?
+* Are there collaborators who work with the individual and can vouch for
+  them.
+
+The build team will review such requests through an issue on the repo. 
+Once agreed the individual will be granted access through the secrets repo
```
Comment (jbergstroem) [inline doc/process/special_access_to_build_resources.md]: Once agreed and voted upon?

File: doc/process/special_access_to_build_resources.md [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+in a way that limits their access to the resources  required (for example
```
Comment (jbergstroem) [inline doc/process/special_access_to_build_resources.md]: nit: two spaces

**Comment by jbergstroem** [comment]
Lots of nits but mostly LGTM!

File: doc/process/special_access_to_build_resources.md (hunk 5 above)
Comment (mhdawson) [inline doc/process/special_access_to_build_resources.md]: I think we want to leave the door open to non-collaborators as some of the options do allow for that.

**Comment by mhdawson** [comment]
added commit to address @jbergstroem comments, incorporate all except for limiting use cases to collaborators only as I think we've left the door open to non-collaborators in some of the cases covered (with more precautions such as re-imaging etc.)

**Comment by mhdawson** [comment]
@jbergstroem let me know if you have any other comments otherwise I'll plan to land early next week.

**Comment by jbergstroem** [comment]
@mhdawson nah looks good lets get it in.

**Comment by mhdawson** [comment]
Landed as 0936e417b7aac392e8b7ba1c39e81a20532b9df3

**Pull request closed without merging**